File: bot-api/src/main/java/gg/mineral/bot/api/entity/effect/PotionEffect.java
package gg.mineral.bot.api.entity.effect;

public interface PotionEffect {
    /**
     * Returns the ID of the potion effect.
     *
     * @return the ID of the potion effect
     */
    int getPotionID();

    /**
     * Returns the type of this potion effect.
     *
     * @return the type of this potion effect
     */
    default PotionEffectType getType() {
        return PotionEffectType.getById(getPotionID());
    }

    /**
     * Returns the amplifier level of this potion effect.
     *
     * @return the amplifier level of this potion effect
     */
    int getAmplifier();

    /**
     * Returns the duration of this potion effect.
     *
     * @return the duration of this potion effect
     */
    int getDuration();
}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/entity/effect/PotionEffect.kt
package gg.mineral.bot.api.entity.effect

interface PotionEffect {
    /**
     * Returns the ID of the potion effect.
     *
     * @return the ID of the potion effect
     */
    val potionID: Int

    val type: PotionEffectType?
        /**
         * Returns the type of this potion effect.
         *
         * @return the type of this potion effect
         */
        get() = PotionEffectType.getById(potionID)

    /**
     * Returns the amplifier level of this potion effect.
     *
     * @return the amplifier level of this potion effect
     */
    val amplifier: Int

    /**
     * Returns the duration of this potion effect.
     *
     * @return the duration of this potion effect
     */
    val duration: Int
}
